src/main/java/it/gov/pagopa/reward/notification/controller/NotificationController.java

```diff
@@ -45,6 +43,34 @@ Mono<Page<RewardExportsDTO>> getExportsPaged(
             @PageableDefault(size = 10) Pageable pageable,
             ExportFilter filters);
 
+    @GetMapping(value = "/organization/{organizationId}/initiative/{initiativeId}/reward/notification/exports/{exportId}")
+    Mono<ExportSummaryDTO> getExport(
+            @PathVariable("organizationId") String organizationId,
+            @PathVariable("initiativeId") String initiativeId,
+            @PathVariable("exportId") String exportId);
+
+    @GetMapping(value = "/organization/{organizationId}/initiative/{initiativeId}/reward/notification/exports/{exportId}/content")
+    Flux<RewardNotificationDTO> getExportNotifications(
+            @PathVariable("organizationId") String organizationId,
+            @PathVariable("initiativeId") String initiativeId,
+            @PathVariable("exportId") String exportId,
+            @PageableDefault(size = 10) Pageable pageable,
+            ExportDetailFilter filters);
+
+    @GetMapping(value = "/organization/{organizationId}/initiative/{initiativeId}/reward/notification/exports/{exportId}/content/paged")
+    Mono<ExportContentPageDTO> getExportNotificationsPaged(
+            @PathVariable("organizationId") String organizationId,
+            @PathVariable("initiativeId") String initiativeId,
+            @PathVariable("exportId") String exportId,
+            @PageableDefault(size = 10) Pageable pageable,
+            ExportDetailFilter filters);
+
+    @GetMapping(value = "/organization/{organizationId}/initiative/{initiativeId}/reward/notification/{notificationId}")
+    Mono<RewardNotificationDetailDTO> getRewardNotification(
+            @PathVariable("organizationId") String organizationId,
+            @PathVariable("initiativeId") String initiativeId,
+            @PathVariable("notificationId") String notificationId);
```

Comment:
due to the need to let filter by externalId, because this application probably will not omit it in response, let's explicitly handle externalId
```suggestion
    @GetMapping(value = "/organization/{organizationId}/initiative/{initiativeId}/reward/notification/byExternalId/{notificationExternalId}")
    Mono<RewardNotificationDetailDTO> getRewardNotification(
            @PathVariable("organizationId") String organizationId,
            @PathVariable("initiativeId") String initiativeId,
            @PathVariable("notificationExternalId") String notificationExternalId);
```

Reply:
thus, inside the inner invocations and models let's explicitly handle the use of externalId.
Also in the models, let's map both fields

Reply:
we will let the application that will resolve the userId into fiscalCode to hide the id and work with the externalId.
Here we will explicitly use both if required